com.unity.netcode.gameobjects/Tests/Runtime/NetworkVariableTests.cs

```diff
@@ -6,9 +6,173 @@
+        }
+
+        protected override IEnumerator OnServerAndClientsConnected()
+        {
+            m_TestObjId = SpawnObject(m_TestObjPrefab, m_ServerNetworkManager).GetComponent<NetworkObject>().NetworkObjectId;
+            yield return null;
+        }
+
+        private void AssertOwnerWritableAreEqualOnAll()
+        {
+            var testObjServer = m_ServerNetworkManager.SpawnManager.SpawnedObjects[m_TestObjId];
+            var testCompServer = testObjServer.GetComponent<NetVarPermTestComp>();
+            foreach (var clientNetworkManager in m_ClientNetworkManagers)
+            {
+                var testObjClient = clientNetworkManager.SpawnManager.SpawnedObjects[m_TestObjId];
+                var testCompClient = testObjClient.GetComponent<NetVarPermTestComp>();
+                Assert.AreEqual(testObjServer.OwnerClientId, testObjClient.OwnerClientId);
+                Assert.AreEqual(testCompServer.OwnerWritable_Position.Value, testCompClient.OwnerWritable_Position.Value);
+                Assert.AreEqual(testCompServer.OwnerWritable_Position.ReadPerm, testCompClient.OwnerWritable_Position.ReadPerm);
+                Assert.AreEqual(testCompServer.OwnerWritable_Position.WritePerm, testCompClient.OwnerWritable_Position.WritePerm);
+            }
+        }
+
+        private void AssertServerWritableAreEqualOnAll()
+        {
+            var testObjServer = m_ServerNetworkManager.SpawnManager.SpawnedObjects[m_TestObjId];
+            var testCompServer = testObjServer.GetComponent<NetVarPermTestComp>();
+            foreach (var clientNetworkManager in m_ClientNetworkManagers)
+            {
+                var testObjClient = clientNetworkManager.SpawnManager.SpawnedObjects[m_TestObjId];
+                var testCompClient = testObjClient.GetComponent<NetVarPermTestComp>();
+                Assert.AreEqual(testCompServer.ServerWritable_Position.Value, testCompClient.ServerWritable_Position.Value);
+                Assert.AreEqual(testCompServer.ServerWritable_Position.ReadPerm, testCompClient.ServerWritable_Position.ReadPerm);
+                Assert.AreEqual(testCompServer.ServerWritable_Position.WritePerm, testCompClient.ServerWritable_Position.WritePerm);
+            }
+        }
+
+        [UnityTest]
+        public IEnumerator ServerChangesOwnerWritableNetVar()
+        {
+            AssertOwnerWritableAreEqualOnAll();
+
+            var testObjServer = m_ServerNetworkManager.SpawnManager.SpawnedObjects[m_TestObjId];
+            var testCompServer = testObjServer.GetComponent<NetVarPermTestComp>();
+
+            var oldValue = testCompServer.OwnerWritable_Position.Value;
+            var newValue = oldValue + new Vector3(Random.Range(0, 100.0f), Random.Range(0, 100.0f), Random.Range(0, 100.0f));
+
+            testCompServer.OwnerWritable_Position.Value = newValue;
+            yield return NetcodeIntegrationTestHelpers.WaitForTicks(m_ServerNetworkManager, 2);
+
+            AssertOwnerWritableAreEqualOnAll();
```

Comment:
if for whatever reason server side Value gets reverted back to it's original value and then client side the Value never changes, this test would still pass :(
with these, I usually prefer to set very specific values and make sure they are the ones expected on arrival.

Reply:
@SamuelBellomo Ok, I see what you are saying... 
@MFatihMAR  So maybe you could do this:
```
private void AssertServerWritableAreEqualOnAll(Vector3 expectedPosition)
{
    var testObjServer = m_ServerNetworkManager.SpawnManager.SpawnedObjects[m_TestObjId];
    var testCompServer = testObjServer.GetComponent<NetVarPermTestComp>();
    foreach (var clientNetworkManager in m_ClientNetworkManagers)
    {
        var testObjClient = clientNetworkManager.SpawnManager.SpawnedObjects[m_TestObjId];
        var testCompClient = testObjClient.GetComponent<NetVarPermTestComp>();
        Assert.AreEqual(testCompServer.ServerWritable_Position.Value, expectedPosition); 
        Assert.AreEqual(testCompServer.ServerWritable_Position.Value, testCompClient.ServerWritable_Position.Value);
        Assert.AreEqual(testCompServer.ServerWritable_Position.ReadPerm, testCompClient.ServerWritable_Position.ReadPerm);
        Assert.AreEqual(testCompServer.ServerWritable_Position.WritePerm, testCompClient.ServerWritable_Position.WritePerm);
    }
}
```
And update your tests to pass in the expected new value to AssertServerWritableAreEqualOnAll?

Reply:
technically that's being caught but the fact that it's hard to imply made me add an extra assert to compare `.Value` against `newValue`